lock-api-objects/src/test/java/com/palantir/lock/watch/LockEventSerializationTest.java

```diff
@@ -0,0 +1,230 @@
+                .build();
+
+        assertThat(deserialize(serialize(largeLockEvent), LockEvent.class)).isEqualTo(largeLockEvent);
+        LockWatchEvent asLockWatchEvent = largeLockEvent;
+        assertThat(deserialize(serialize(asLockWatchEvent), LockWatchEvent.class))
+                .isEqualTo(asLockWatchEvent);
+
+        OldLockEvent largeLockEventWithoutMetadata = ImmutableOldLockEvent.builder()
+                .sequence(largeLockEvent.sequence())
+                .lockDescriptors(largeLockEvent.lockDescriptors())
+                .lockToken(largeLockEvent.lockToken())
+                .build();
+        assertThat(deserialize(serialize(largeLockEvent), OldLockEvent.class)).isEqualTo(largeLockEventWithoutMetadata);
+    }
+
+    @Test
+    public void shufflingLockDescriptorsOnTheWireIsDetectedByClient() {
+        assertThatThrownBy(() ->
+                        deserialize(Files.readString(getJsonPath("baseline-with-shuffled-locks")), LockEvent.class))
+                .rootCause()
+                .isInstanceOf(SafeIllegalArgumentException.class)
+                .hasMessageStartingWith("Key list integrity check failed");
+    }
+
+    private static <T> void assertSerializedEquals(T object, String jsonFileName) {
+        try {
+            Path path = getJsonPath(jsonFileName);
+            if (REWRITE_JSON_BLOBS) {
+                SERVER_MAPPER.writeValue(path.toFile(), object);
+            }
+            String serialized = serialize(object);
+            assertThat(VERIFYING_MAPPER.readTree(serialized))
+                    .as("Serialization yields identical JSON representation")
+                    .isEqualTo(VERIFYING_MAPPER.readTree(Files.readString(path)));
+        } catch (Throwable t) {
+            throw new RuntimeException(t);
+        }
+    }
+
+    private static <T> void assertDeserializedEquals(String jsonFileName, T object, Class<T> clazz) {
+        try {
+            assertThat(deserialize(Files.readString(getJsonPath(jsonFileName)), clazz))
+                    .isEqualTo(object);
+        } catch (Throwable t) {
+            throw new RuntimeException(t);
+        }
+    }
+
+    private static <T> String serialize(T object) {
+        try {
+            return SERVER_MAPPER.writeValueAsString(object);
+        } catch (Throwable t) {
+            throw new RuntimeException(t);
+        }
+    }
+
+    private static <T> T deserialize(String json, Class<T> clazz) {
+        try {
+            return CLIENT_MAPPER.readValue(json, clazz);
+        } catch (Throwable t) {
+            throw new RuntimeException(t);
+        }
+    }
+
+    // This is a copy of the LockEvent class before metadata was added
+    @Value.Immutable
+    @JsonSerialize(as = ImmutableOldLockEvent.class)
+    @JsonDeserialize(as = ImmutableOldLockEvent.class)
+    @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+    @JsonTypeName(LockEvent.TYPE)
+    public interface OldLockEvent {
```

Comment:
This is property not a good practice, but I wanted to be sure that we are backwards compatible with this change (or rather that clients are forward-compatible).